src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,79 @@
 package core.basesyntax;
 
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.List;
+
 public class WorkWithFile {
+    public static final String COMMA = ",";
+    public static final String SUPPLY = "supply";
+    public static final String BUY = "buy";
+    public static final String RESULT = "result";
+    private int supplyAmount = 0;
+    private int buyAmount = 0;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        StringBuilder stringBuilder = new StringBuilder();
+
+        supplyAmount = 0;
+        buyAmount = 0;
+        File fromFile = new File(fromFileName);
+        final File toFile = new File(toFileName);
+
+        calculateReport(fromFile, stringBuilder);
+
+        writeIntoFile(toFile, stringBuilder);
+    }
+
+    private void calculateReport(File fromFile, StringBuilder stringBuilder) {
+
+        readFromFile(fromFile);
+
+        stringBuilder.append(SUPPLY).append(COMMA).append(supplyAmount)
+                .append(System.lineSeparator())
+                .append(BUY).append(COMMA).append(buyAmount)
+                .append(System.lineSeparator())
+                .append(RESULT).append(COMMA).append(supplyAmount - buyAmount);
+    }
+
+    private void readFromFile(File fromFile) {
+        List<String> fileData;
+        try {
+            fileData = Files.readAllLines(fromFile.toPath());
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read from file", e);
+        }
+
+        for (String data : fileData) {
+            String[] values = data.split(COMMA);
+
+            if (values.length != 2) {
+                continue;
+            }
+
+            String operation = values[0];
+            String amountString = values[1];
+
+            try {
+                int value = Integer.parseInt(amountString);
+                if (operation.equals(BUY)) {
+                    buyAmount += value;
+                } else {
+                    supplyAmount += value;
+                }
+            } catch (NumberFormatException e) {
+                throw new RuntimeException("Invalid number format in file: " + amountString);
```

Comment:
When throwing a new RuntimeException for a `NumberFormatException`, it would be helpful to include the `e` as a cause in the new exception to preserve the stack trace and original exception message: `throw new RuntimeException("Invalid number format in file: " + amountString, e);`